### DISCUSSION TITLE
Enable installing Caddy with plugins

### DISCUSSION THREAD

**Discussion by nehalvpatel** [Homebrew/discussions#886] [Everyday usage] [answered]
Is there any way to install [Caddy](https://formulae.brew.sh/formula/caddy) with plugins? The current version of the formulae installs a clean, no-plugins build.

Their [download page](https://caddyserver.com/download) lets you generate arbitrary builds of Caddy with any combination of plugins. It also supports M1 builds, but they haven't made any announcement about it as far as I can tell.

They also have a way for you to build from source while including plugins with [xcaddy](https://caddyserver.com/docs/build).

I've installed Caddy using Homebrew, because I like that it creates and manages the LaunchAgent for me. Because I need the Cloudflare plugin, I've replaced the binary at `/opt/homebrew/bin/caddy` (which is actually `/opt/homebrew/Cellar/caddy/2.3.0/bin/caddy`) with a custom-built binary from the download page. This binary includes the Cloudflare plugin.

Of course, my custom binary will get wiped the next time Caddy pushes a tagged release. This is why I wanted to bring this up. I couldn't find any previous discussion about this, so I've started this one here.

I'm not sure how the ability to include plugins could be handled UX wise, but I think it would be a useful feature.

**Comment by SMillerDev** [comment, marked as answer]
Basically we don't support formula with plugins unless the plugins are separate software. For Caddy I believe there is no way to make the plugins optional so this wouldn't be allowed in homebrew/core.

But you can always consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Reply by nehalvpatel** [reply]
That seems like it'll be a steep learning curve, but I agree with your reasoning, for sure. Is there any existing tap that uses this plugin build structure that you could point me to?

**Reply by SMillerDev** [reply]
Not that I know of but since taps are just github repositories, I don't know that many. There is also not a set "plugin build structure". I'm suggesting you copy the Caddy formula to your tap and just enable the plugins you want there.

**Reply by nehalvpatel** [reply]
Ah, I see what you mean now. Thanks, I'll give that a shot.
